File: frontend/src/types/task.ts
export interface ManualTask {
  id: string;
  project_id?: string | null;
  task_type: string;
  status: string;
  progress: number;
  input_json?: Record<string, unknown> | null;
  output_json?: Record<string, unknown> | null;
  error_message?: string | null;
}

File: backend/app/schemas/task.py
from pydantic import BaseModel, Field
from app.schemas.common import Timestamped


class TaskRead(Timestamped):
    project_id: str | None
    task_type: str
    status: str
    progress: int
    input_json: dict | None = None
    output_json: dict | None = None
    error_message: str | None = None


class TaskCreated(BaseModel):
    task_id: str
    status: str


class TranslateRequest(BaseModel):
    target_language: str
    page_ids: list[str] | None = None
    use_terms: bool = True
    keep_layout: bool = True


class QualityCheckRequest(BaseModel):
    target_language: str
    page_ids: list[str] | None = None


class ExportPdfRequest(BaseModel):
    language: str = "source"
    page_ids: list[str] | None = None
